### PR TITLE
Upgrade pyright CI

### PR DESCRIPTION
**Pull request by jwortmann**
After installing Python 3.12, running `tox` in this repo is currently broken on my Windows PC. An upgrade of pyright seems to fix it.

> Ignore new type errors

It's unclear to me how to fix those, but if anyone knows it feel free to write or push directly:

promise

sessions

### CODE REVIEW

**Comment by rchl** [comment]
> It's unclear to me how to fix those, but if anyone knows it feel free to write or push directly:

Pyright's author helpful as always: https://github.com/microsoft/pyright/discussions/6301

File: plugin/core/sessions.py [hunk 1]
```diff
@@ -1489,7 +1489,7 @@ def _handle_initialize_success(self, result: InitializeResult) -> None:
             self._workspace_folders = self._workspace_folders[:1]
         self.state = ClientStates.READY
         if self._plugin_class is not None:
-            self._plugin = self._plugin_class(weakref.ref(self))  # type: ignore  # TODO fix me
+            self._plugin = self._plugin_class(weakref.ref(cast(Session, self)))
```
Comment (rchl) [inline plugin/core/sessions.py:1492]: This type issue is also a bug:

https://github.com/microsoft/pyright/discussions/6334
https://github.com/microsoft/pyright/issues/6335

Comment (jwortmann) [reply plugin/core/sessions.py:1492]: So we should rather wait for the release of an unbugged Pyright version, I guess?

Comment (rchl) [reply plugin/core/sessions.py:1492]: yeah

**Pull request merged**